docs/babelconfigjs.md

```diff
@@ -0,0 +1,97 @@
+---
+title: babel.config.js
+id: babelconfigjs
+---
+
+## Lookup behavior
+
+The lookup behavior is simple, Babel will only look for a `babel.config.js` in the root directory (where your package.json is by default). You can configure if needed, see [Specifying the root folder](babelconfigjs.md#specifying-the-root-folder).
+
+## Merging behavior
+
+Unlike the other formats (see [configuration guide](condiguration.md)), Babel won't try to merge configurations.
+
+> Note that the configuration defined in `babel.config.js` is applied in your entire project.
+
+## Conditional configuration
+
+Since it's just a regular JavaScript file, you can write any arbitrary conditions before returning the configuration, for example:
+
+```js
+module.exports = function () {
+  const plugins = [ ... ];
+
+  if (true === false) {
+    plugins.push(...);
+  }
+
+  return { plugins };
+}
+```
+
+### Environment
+
+A conditional configuration is useful when you want to use certain plugins/presetns based on the current environement, for example:
+
+```js
+module.exports = function () {
+  const plugins = [ ... ];
+
+  if (process.env["REMOVE_DEBUG"] === 1) {
+    plugins.push("babel-plugin-remove-console-debug");
+  }
+
+  return { plugins };
+}
+```
+
+You can invoke it using the CLI:
+
+```sh
+REMOVE_DEBUG=1 babel file.js
+```
+
+## API
+
+Babel will pass an object as first argument, called `api` here.
+
+```js
+module.exports = function (api) {
+  const babelEnv = api.env();
+
+  return { ... };
+}
+```
+
+### `api.env()`
+
+Returns Babel's environement, you can configure it using the `BABEL_ENV` environement variable.
```

Comment:
2x`environment`

Reply:
Do you have a suggestion?

Reply:
Oh, I meant there's 2 typos in `environment` here

Reply:
Yes but I couldn't find a better wording for that, do you have a suggestion?

Reply:
This actually sounds clear to me even with the repetitiveness.